couchdb/couchdb.go

```diff
@@ -154,12 +166,57 @@ func DeleteDB(dbprefix, doctype string) error {
 }
 
 // ResetDB destroy and recreate the database for a doctype
-func ResetDB(dbprefix, doctype string) error {
-	err := DeleteDB(dbprefix, doctype)
-	if err != nil {
-		return err
+func ResetDB(dbprefix, doctype string) (err error) {
+	err = DeleteDB(dbprefix, doctype)
+	if err == nil {
+		return CreateDB(dbprefix, doctype)
+	}
+	return
```

Comment:
And I think it's a golang convention to have the good case path in the final return, and errors in the if cases.